app/main.py

```diff
@@ -1,2 +1,242 @@
+from __future__ import annotations
+
+import copy
+import dataclasses
+from typing import Any, Callable, Iterable, Hashable
+
+
+@dataclasses.dataclass
+class Node:
+    key: Hashable
+    k_hash: int
+    value: Any
+
+
 class Dictionary:
-    pass
+
+    def __init__(self) -> None:
+        # added mainly to keep track of input order, iteration
+        self._keys = []
+        self._values = []
+
+        self._initial_capacity = 8
+        self._load_factor = 2 / 3
+
+        self._capacity = self._initial_capacity
+        self._hash_table = [[] for _ in range(self._capacity)]
+
+    # didn't add formating for different types due to time complexity
+    def __str__(self) -> str:
+        pairs = []
+        for key, value in zip(self._keys, self._values):
+            pairs.append(f"{key}: {value}")
+        result = "{" + ", ".join(pairs) + "}"
+        return result
+
+    def __getitem__(self, key: Hashable) -> Any:
+        return self._data_handler(key)
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+
+        if self.get_load():
+            self._data_handler(key, value)
+        else:
+            self.resize()
+            self.__setitem__(key, value)
+
+    def __len__(self) -> int:
+        return sum(1 for item in self._hash_table if item)
```

Comment:
Why not have an extra attribute `size` that you will use to keep track of occupied cells in hash table? Then your `len` operation will take O(1) instead of O(n)